crates/sui-core/src/authority.rs

```diff
@@ -1227,12 +1233,163 @@ impl AuthorityState {
                 .move_calls()
                 .iter()
                 .map(|mc| (mc.package.0, mc.module.clone(), mc.function.clone())),
+            changes,
             seq,
             digest,
             timestamp_ms,
         )
     }
 
+    fn process_object_index(
+        &self,
+        effects: &SignedTransactionEffects,
+    ) -> Result<ObjectIndexChanges, SuiError> {
+        let modified_at_version = effects
+            .modified_at_versions
+            .iter()
+            .cloned()
+            .collect::<HashMap<_, _>>();
+
+        let mut deleted_owners = vec![];
+        let mut deleted_dynamic_fields = vec![];
+        for (id, _, _) in &effects.deleted {
+            let Some(old_version) = modified_at_version.get(id) else{
+                error!("Error processing object owner index for tx [{}], cannot find modified at version for deleted object [{id}].", effects.transaction_digest);
+                continue;
+            };
+            match self.get_owner_at_version(id, *old_version)? {
+                Owner::AddressOwner(addr) => deleted_owners.push((addr, *id)),
+                Owner::ObjectOwner(object_id) => {
```

Comment:
good catch! this will save some unnecessary seek and delete ops on db

Reply:
Actually after looking at the code again (this was done quite a while ago 😅), it will require an extra DB read to retrieve the object type because TransactionEffects only contains ObjectRef, which will be more expensive because it will deserialise the whole `Object` from storage. We could do this without the db read after this [PR](https://github.com/MystenLabs/sui/pull/6636) which adds ObjectType to the effects.

I will add a TODO for now.